MapboxCoreNavigation/RouteController.swift

```diff
@@ -315,6 +335,40 @@ open class RouteController: NSObject {
     }
     
     /**
+     Checks whether a new location is quality by comparing it to the previous location.
+     
+     We're attempting to throw out coordinates that have a large difference in location or time.
+     */
+    func isQualityLocation(location: CLLocation?) -> Bool {
+        
+        // Any time we return in this func, reset the values.
+        defer {
+            if let location = location {
+                if let lastLocation = previousKnownLocation {
+                    previousKnownLocation?.locationDelta = location.distance(from: lastLocation.location)
+                    previousKnownLocation?.durationDelta = location.timestamp.timeIntervalSince(lastLocation.location.timestamp)
+                }
+                
+                previousKnownLocation?.location = location
+            }
+        }
+        
+        guard let lastLocation = previousKnownLocation, let location = rawLocation else {
+                return false
+        }
+        
+        guard location.distance(from: lastLocation.location) <= lastLocation.locationDelta * 3 else {
```

Comment:
If the user enters a 500-foot-long tunnel at 60 mph, does the next location get tossed out as a poor-quality location?

Reply:
Yes, this is the point. We basically throw out the next location update until locations start streaming in at a regular interval again.

Reply:
Note that this may impact the following use cases:

* The user has just exited the tunnel and there’s an intersection or exit right after.
* The user has just speed up after a traffic light turns green.

Reply:
At worst, we'll be just 1s behind once locations start streaming in, which is better than rerouting and confusing the user.